### PR TITLE
Add strictOutput matching to plan matching framework

### PR DESCRIPTION
**Pull request by ebd2**
Some tests require being able to ensure that the set of outputs for a node is
exactly what is expected, rather than a superset.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/StrictAssignedSymbolsMatcher.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        return expected.build();
+    }
+
+    public static Function<PlanNode, Set<Symbol>> actualAssignments()
+    {
+        return node -> ((ProjectNode) node).getAssignments().keySet();
+    }
+
+    public static Function<PlanNode, Set<Symbol>> actualSubqueryAssignments()
```
Comment (cawallin) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/StrictAssignedSymbolsMatcher.java]: This is unused

Comment (ebd2) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/StrictAssignedSymbolsMatcher.java]: @sopel39 will need it in the very near future.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanMatchingFramework.java [hunk 2]
```diff
@@ -69,6 +73,24 @@ public void testOutputSameColumnMultipleTimesWithOtherOutputs()
     }
 
     @Test
+    public void testStrictOutput()
+    {
+        assertMinimallyOptimizedPlan("SELECT orderkey, extendedprice FROM lineitem",
+                strictOutput(ImmutableList.of("ORDERKEY", "EXTENDEDPRICE"),
+                        tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey",
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanMatchingFramework.java]: Can you also use `strictOutput` after intermediate nodes after `tableScale`? Let's say I want to test strict output of pipeline:
`project -> filter -> tableScan`, where `project` output symbols should be strictly symbols from `tableScan`?

Comment (ebd2) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanMatchingFramework.java]: That should work. As long as you aren't introducing any new symbols in the project node, you should be able to use strictOutput with the aliases you already have bound.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/StrictSymbolsMatcher.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @Override
+    protected Set<Symbol> getExpectedSymbols(PlanNode node, Session session, Metadata metadata, SymbolAliases symbolAliases)
+    {
+        return expectedAliases.stream()
+                .map(symbolAliases::get)
+                .map(Symbol::from)
+                .collect(ImmutableCollectors.toImmutableSet());
+    }
+
+    public static Function<PlanNode, Set<Symbol>> actualOutputs()
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/StrictSymbolsMatcher.java]: put static methodst at the top of the class

File: presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/StrictAssignedSymbolsMatcher.java (hunk 1 above)
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/StrictAssignedSymbolsMatcher.java]: can you please add a factory method in `PlanMatchPatter` which is using this?

File: presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanMatchingFramework.java (hunk 2 above)
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanMatchingFramework.java]: OK I will try in actual code and see if I manage to port tests

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**